packages/core/admin-client/src/services/ServiceConfigs.ts

```diff
@@ -14,133 +14,123 @@
  * limitations under the License.
  */
 
-import { Codec } from "mongodb-stitch-core-sdk";
-
-export interface ServiceConfig {
-  readonly name: string;
-  readonly type: string;
-  readonly config: object;
-
-  readonly configCodec?: Codec<any>;
-}
-
-export class ServiceConfigCodec implements Codec<ServiceConfig> {
-  public decode(from: any): ServiceConfig {
-    const type = from.type;
-    let config: object = from.config;
-    if (type === "twilio") {
-      config = new TwilioConfigCodec().decode(config);
-    }
-
-    return {
-      config,
-      name: from.name,
-      type: from.type
-    };
-  }
-
-  public encode(from: ServiceConfig): object {
-    return {
-      config: from.configCodec
-        ? from.configCodec.encode(from.config)
-        : from.config,
-      name: from.name,
-      type: from.type
-    };
+import { Rule } from "..";
+import { jsonProperty, Type } from "../JsonMapper";
+import { AwsS3Rule, AwsSesRule, HttpRule, MongoDbRule, TwilioRule } from "./rules/RulesResources";
+
+export type Config = any;
```

Comment:
Sort of like an empty interface in Java– I use it later on with generics so that getting the resource for a Service gives you type safe access to the correct configs and rule sets.

Reply:
kk as long as it doesn't just let you do anything type unsafely with a `Config` type I'm fine with it